source/agora/common/crypto/Schnorr.d

```diff
@@ -230,31 +241,59 @@ public Signature sign (T) (
 public bool verify (T) (const ref Point X, const ref Signature sig, auto ref T data)
     nothrow @nogc @trusted
 {
-    Sig s = Sig.fromBlob(sig);
-    // Compute the challenge and reduce the hash to a scalar
-    Scalar c = hashFull(Message!T(X, s.R, data));
-    // Compute `R + c*X`
-    Point RcX = s.R + (c * X);
-    /// Compute `s.G`
-    auto S = s.s.toPoint();
-    return S == RcX;
+    try {
+        Sig s = Sig.fromBlob(sig);
+        if (s.s.isValid()) {
+            /// Compute `s.G`
+            auto S = s.s.toPoint();
+            // Compute the challenge and reduce the hash to a scalar
+            Scalar c = hashFull(Message!T(X, s.R, data));
+            // Compute `R + c*X`
+            Point RcX = s.R + (c * X);
+            return S == RcX;
+        } else {
+            return false;
+        }
+    } 
+    catch(Exception e) 
+    {
+        debug { import std.stdio : writeln; try { writeln("verify signature exception: %s", e.message); } catch (Exception) {} }
+        return false;
```

Comment:
I'm not sure this is correct. If we have explicitly handled the case where the scalar is invalid (`s.s.isValid()`), then we assume that `.toPoint()` will never throw. Therefore if it does throw it's a coding bug - and it should assert.

Secondly, we should not wrap the entire body of the function in a try/catch, but only the code parts which actually throw. So we could write something like:

```D
Point S = void;
try { S = s.s.toPoint(); } catch (Exception ex) { assert(0, ex.msg); }
...
```

Reply:
Updated to return early but in the end there is no need for the try catch as the toPoint would assert anyway.